### PR TITLE
Fixed vertical alignment of FAB-button

### PR DESCRIPTION
**Pull request by david8611**
This should fix the vertical alignment of the FAB-button on IOS. Now the bottom placement is based on the position of the bottom navbar. This is checked dynamically. Due to the Navbar not being rendered in the bottomNavigation.jsx I decided to move the rendering of the FAB.button to the homescreen where it will be rendered in the portal across the screen children in the navbar (Authors and Home screen currently). This adjustment made it possible to obtain the height of the rendered navbar dynamically instead of using an unrendered position approach.

### CODE REVIEW

File: App/components/BottomNavigation.jsx [hunk 1]
```diff
@@ -1,62 +1,40 @@
...
-        <FAB
-          visible={isFocused && !isOpen}
-          icon="plus"
-          style={styles.fab}
-          large
-          onPress={() => console.log('Pressed')}
-          color="#fff"
-        />
-      </Portal>
+      <Portal.Host>
```
Comment (david8611) [inline App/components/BottomNavigation.jsx:14]: Wrapped the Portal host to only cover the screens in the navbar. The default host is embedded in the PaperProvider which is the component furthest up the hierarcy. This means that the FAB-button won'tt be rendered outside the navBar.

**Comment by david8611** [comment]
While testing the alignment of the FAB-button, try also the swiping gestures of the drawer. The swiping gestures work fine on my android phone at the moment.

**Comment by editf** [comment]
The swiping gestures work fine for me!
The positioning of the FAB looks a bit different depending on the size of the screen.
On an iphone simulator: 


On my phone (old iphone SE) with a smaller screen:

It's a bit unrelated but as we can see the home screen doesn't actually fit on my smaller screen and can't be scrolled to access the last buttons, but that should be a separate PR.

**Comment by david8611** [comment]
> The swiping gestures work fine for me!
> The positioning of the FAB looks a bit different depending on the size of the screen.
> On an iphone simulator:
> 
> 
> On my phone (old iphone SE) with a smaller screen:
> 
> It's a bit unrelated but as we can see the home screen doesn't actually fit on my smaller screen and can't be scrolled to access the last buttons, but that should be a separate PR.

Ok, I Will try to subtract a finite value from the actual Height coordinate instead of dividing it (due to the different values on different devices). Hopefully that should fix it. If not, I would suggest we fix the bottom navbar size so that it is the same across all devices.

Do you have any other suggestions?

**Comment by editf** [comment]
> 
> 
> > The swiping gestures work fine for me!
> > The positioning of the FAB looks a bit different depending on the size of the screen.
> > On an iphone simulator:
> > 
> > On my phone (old iphone SE) with a smaller screen:
> > 
> > It's a bit unrelated but as we can see the home screen doesn't actually fit on my smaller screen and can't be scrolled to access the last buttons, but that should be a separate PR.
> 
> Ok, I Will try to subtract a finite value from the actual Height coordinate instead of dividing it (due to the different values on different devices). Hopefully that should fix it. If not, I would suggest we fix the bottom navbar size so that it is the same across all devices.
> 
> Do you have any other suggestions?
> As far as the screen size is concerned I think it is better to fix that in a seperate PR.

I'm sorry, I haven't really had the time to look into it yet. I could try during the weekend. I'm wondering if it might possibly have something to do with the black bar-thingy that is part of the bottom screen on the newer iphone? Is that extra space part of the calculations or not?

**Comment by david8611** [comment]
> > > The swiping gestures work fine for me!
> > > The positioning of the FAB looks a bit different depending on the size of the screen.
> > > On an iphone simulator:
> > > 
> > > On my phone (old iphone SE) with a smaller screen:
> > > 
> > > It's a bit unrelated but as we can see the home screen doesn't actually fit on my smaller screen and can't be scrolled to access the last buttons, but that should be a separate PR.
> > 
> > 
> > Ok, I Will try to subtract a finite value from the actual Height coordinate instead of dividing it (due to the different values on different devices). Hopefully that should fix it. If not, I would suggest we fix the bottom navbar size so that it is the same across all devices.
> > Do you have any other suggestions?
> > As far as the screen size is concerned I think it is better to fix that in a seperate PR.
> 
> I'm sorry, I haven't really had the time to look into it yet. I could try during the weekend. I'm wondering if it might possibly have something to do with the black bar-thingy that is part of the bottom screen on the newer iphone? Is that extra space part of the calculations or not?

Yes I believe it should be. Can you try it now after I changed the value to a subtraction?
Otherwise I think we have to fix the size of the navbar..

**Comment by editf** [comment]
It works now! 
Skärmavbild 2021-02-18 kl  12 39 26

**Review by editf — APPROVED** [review]

**Pull request merged**